examples/guess-the-sketch/src/app.py

```diff
@@ -76,106 +83,114 @@ def agones_health():
-
-        similarity_response_cur = requests.post(f'http://genai-api.genai.svc/genai/text', headers=headers, json=payload_similarity)
-
-        # logger.info(f'Similarity response: {similarity_response_cur.text}')
-        similarity_prompt_cur_without_quota = similarity_response_cur.text.strip('"')
-        logger.info(f'Similarity response without quota: {similarity_prompt_cur_without_quota}')
-
-        parts = similarity_prompt_cur_without_quota.split(",")  # Split the string by commas
-
-        # Extract numbers and convert to float
-        score1 = float(parts[0].split()[-1])  
-        score2 = float(parts[1].split()[-1]) 
-
-        logger.info(f'num1: {score1}')
-        logger.info(f'num2: {score2}')
-
-        if score1 > score2:
-            emit('winner', {'message': 'You won!'}, room=request.sid)
-            emit('winner', {'message': 'You lost!'}, room=partner_id)
-        elif score1 < score2:
-            emit('winner', {'message': 'You lost!'}, room=request.sid)
-            emit('winner', {'message': 'You won!'}, room=partner_id)
-        else:
-            emit('winner', {'message': 'It\'s a tie!'}, room=request.sid)
-            emit('winner', {'message': 'It\'s a tie!'}, room=partner_id)
-
-        del pending_requests[request.sid]
-        del pending_requests[partner_id]
-
-@socketio.on('message')
-def handle_message(message):
-    request_id = request.sid
-    logger.info('Received request %s', request_id)
-    logger.info('socket id %s', request.sid)
-    logger.info('Message: %s', message)
-    logger.info('Pending requests: %s', pending_requests)
-
-    # TODO ... (Timeout logic - Consider a background task or database) ...
-
-    pending_requests[request_id] = {
-        'timestamp': time.time(),
-        'partner': None,
-        'socket_id': request.sid,
-        'message': message,
-        'partner_message': None,
-        'guess': None
+def handle_message(data):
+    player_id = request.sid
+    message = data['message']
+    oppontent_id = data['opponentId']
+    round = int(data['round'])
+
+    logger.debug('Received guess %s from player %s for oppontent %s: %s', str(round), player_id, oppontent_id, message)
+
+    # Store the guess
+    if player_guess.get(player_id) is None:
+        player_guess[player_id] = {}
+    player_guess[player_id][round] = {
+        'guess': message
+    }
+    
+    # Send next picture which is generated by the opponent's prompt
+    if round != game_round:
+        next_round = round + 1
+        while player_prompt[oppontent_id][next_round]['picture'] is None:
+            time.sleep(0.1)
+        emit('guess_sketch_response', {'image': player_prompt[oppontent_id][next_round]['picture'], 'prompt': player_prompt[oppontent_id][next_round]['prompt'], 'opponentId': oppontent_id, 'round': next_round}, room=player_id)
+        logger.debug('Sent image %s to %s', str(next_round), player_id)
+
+    # Generate and store the picture for the guess
```

Comment:
this section might be better off in a separate function, since it looks duplicated in a couple of places in the file (and then switching between vertex and SD would be a single change instead of two changes).